src/components/SignUpDialogueBox.tsx

```diff
@@ -1,24 +1,89 @@
 // React imports
-import React, { useState } from "react";
+import React, { useState, SyntheticEvent } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useMutation, useApolloClient } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
 
-import UploadFile from "./UploadFile";
+import UploadFile from './UploadFile';
 
 // Type definitions
 type DialogueProps = {
   isVolunteer: boolean;
   setIsVolunteer: any;
 };
 
+const REGISTER_VOLUNTEER = gql`
+  mutation registerVolunteer(
+    $username: String!
+    $password: String!
+    $confirmPassword: String!
+  ) {
+    registerVolunteer(
+      username: $username
+      password: $password
+      confirmPassword: $confirmPassword
+    ) {
+      ok
```

Comment:
Is it normal to request these things back?  Doesn't graphql have some protocol in place for errors? (I  have no idea, I just would have assumed)

Reply:
I did this for some easier to read error messages that we can display. Like validation stuff. Errors in [useMutation ](https://www.apollographql.com/docs/react/data/mutations/) hook confuse me. In the useQuery hook I have access to errors.